src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -55,6 +58,13 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             $this->logger->info('Received message {class}', $context);
         } else {
             $shouldDispatchEvent = true;
+
+            if (false === $this->throwNoRouting
+                && 0 === iterator_count($this->sendersLocator->getSenders($envelope))
```

Comment:
`getSenders` returns a \Generator, not an iterable.

Reply:
Instead of calling iterator_count, we should add a check after the foreach below, to prevent iterating twice over the iterator.